skfda/preprocessing/smoothing/_basis.py

```diff
@@ -0,0 +1,520 @@
+            data as an FDataGrid, like the other smoothers. If ``True`` returns
+            a FDataBasis object.
+
+    Examples:
+        By default, the data is converted to basis form without smoothing:
+
+        >>> import numpy as np
+        >>> import skfda
+        >>> t = np.linspace(0, 1, 5)
+        >>> x = np.sin(2 * np.pi * t) + np.cos(2 * np.pi * t)
+        >>> x
+        array([ 1.,  1., -1., -1.,  1.])
+
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='cholesky', return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.  , 0.71, 0.71]])
+
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='qr', return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[-0.  , 0.71, 0.71]])
+
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='matrix', return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.  , 0.71, 0.71]])
+        >>> smoother.hat_matrix().round(2)
+        array([[ 0.43,  0.14, -0.14,  0.14,  0.43],
+               [ 0.14,  0.71,  0.29, -0.29,  0.14],
+               [-0.14,  0.29,  0.71,  0.29, -0.14],
+               [ 0.14, -0.29,  0.29,  0.71,  0.14],
+               [ 0.43,  0.14, -0.14,  0.14,  0.43]])
+
+        If the smoothing parameter is set to something else than zero, we can
+        penalize approximations that are not smooth enough using a linear
+        differential operator:
+
+        >>> from skfda.misc import LinearDifferentialOperator
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='cholesky',
+        ...                smoothing_parameter=1,
+        ...                penalty=LinearDifferentialOperator(weights=[3, 5]),
+        ...                return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.18,  0.07,  0.09]])
+
+        >>> from skfda.misc import LinearDifferentialOperator
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='qr',
+        ...                smoothing_parameter=1,
+        ...                penalty=LinearDifferentialOperator(weights=[3, 5]),
```

Comment:
I think the differential operator is not being built properly (or printed). 

```python
>>> smoother.penalty
LinearDifferentialOperator(
    nderiv=1,
    bwtlist=[
    FDataBasis(
        basis=Constant(domain_range=[array([0, 1])], nbasis=1),
        coefficients=[[3]],
        dataset_label=None,
        axes_labels=None,
        extrapolation=None,
        keepdims=False)]
    )
```

Reply:
It was the `__repr__` method. Changed also in 73ec1d07ea2d6aae469481a3ed30eaa426f4b273.